src/app/server/JointFabricDatastore.cpp

```diff
@@ -345,40 +494,72 @@ JointFabricDatastore::UpdateGroup(const Clusters::JointFabricDatastore::Commands
 {
     size_t index = 0;
     // Check if the group ID exists in the datastore
-    VerifyOrReturnError(IsGroupIDInDatastore(commandData.groupID, index) == CHIP_NO_ERROR,
-                        CHIP_ERROR_IM_STATUS_CODE_CONSTRAINT_ERROR);
+    VerifyOrReturnError(IsGroupIDInDatastore(commandData.groupID, index) == CHIP_NO_ERROR, CHIP_ERROR_CONSTRAINT);
 
     if (commandData.groupCAT.ValueOr(0) == kAdminCATIdentifier || commandData.groupCAT.ValueOr(0) == kAnchorCATIdentifier)
     {
         // If the group is an AdminCAT or AnchorCAT, we cannot update it
-        return CHIP_ERROR_IM_STATUS_CODE_CONSTRAINT_ERROR;
+        return CHIP_ERROR_CONSTRAINT;
     }
 
     // Update the group entry with the new data
     if (commandData.friendlyName.IsNull() == false)
     {
+        if (mGroupInformationEntries[index].friendlyName.data_equal(commandData.friendlyName.Value()) == false)
+        {
+            // TODO: Spec needs to be updated regarding this step (unsure what to do)
+        }
+        // Update the friendly name
         mGroupInformationEntries[index].friendlyName = commandData.friendlyName.Value();
     }
     if (commandData.groupKeySetID.IsNull() == false)
     {
+        if (mGroupInformationEntries[index].groupKeySetID != commandData.groupKeySetID.Value())
+        {
+            // If the groupKeySetID is being updated, we need to ensure that the new key set exists
```

Comment:
```suggestion
            // TODO: create new NodeKeySetEntry for each node in this group, remove old ones.
            // If the groupKeySetID is being updated, we need to ensure that the new key set exists
```

Reply:
This is already being handled in the lines below the comment.

Reply:
Could you elaborate? The UpdateGroup command doesn't include a NodeID.

Reply:
Added a TODO statement as recommended.